[Auditd Manager] Add Session Data option

opauloh opened the pull request:
## Proposed commit message

- Updating Auditd Manager manifest to include Session Data option starting from Kibana version 8.16.0.
- Updated logic in the `hbs` file to append Session Data Audit Rules and Session Data Processors when Session Data is selected.
- Updated documentation to include both manual and Toggle Switch options

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).



## How to test this PR locally

To test these changes, go to Kibana -> Integrations -> Create new integration -> upload it as a .zip; and upload the following package:

[auditd_manager-1.18.0.zip](https://github.com/user-attachments/files/17333356/auditd_manager-1.18.0.zip)


**Please note**. This Kibana [PR](https://github.com/elastic/kibana/pull/195159) should first be merged to test this integration.

## Related issues

- Closes https://github.com/elastic/security-team/issues/10330

## Screenshots

Session Data Switcher

<img width="1657" alt="image" src="https://github.com/user-attachments/assets/4a6480f0-27a6-4d6f-9528-a9c43b1e7553">

elastic-agent .yml

<img width="792" alt="image" src="https://github.com/user-attachments/assets/0334a4ce-b5df-4059-a3aa-a0e91e0d47ce">


Docs

<img width="1652" alt="image" src="https://github.com/user-attachments/assets/71e556cd-6e8f-4b3a-9666-0041bb56c59b">

elasticmachine commented:
Pinging @elastic/sec-linux-platform (Team:Security-Linux Platform)

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/16821)
* Commit: f1ab7f92ec70452727ce9b284dc6425af810ea41

### Failed CI Steps
* [Check integrations auditd_manager](https://buildkite.com/elastic/integrations/builds/16821#01925a5a-ac6b-4cbf-96ab-af65f48efd3b)

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/16821","number":16821,"commit":"f1ab7f92ec70452727ce9b284dc6425af810ea41"}],"number":16821}
buildkite-pr-comment-->

nick-alayil reviewed (approved):
LGTM

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/16929)
* Commit: 87b75b09a8ad7969eb4a3033ee7c7c80b7e6a835

### Failed CI Steps
* [Check integrations auditd_manager](https://buildkite.com/elastic/integrations/builds/16929#01926d82-f1f7-4d0f-8e82-92e1e0d1cc88)

### History
* :broken_heart: [Build #16821](https://buildkite.com/elastic/integrations/builds/16821) failed f1ab7f92ec70452727ce9b284dc6425af810ea41

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/16929","number":16929,"commit":"87b75b09a8ad7969eb4a3033ee7c7c80b7e6a835"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/16821","number":16821,"commit":"f1ab7f92ec70452727ce9b284dc6425af810ea41"}],"number":16929}
buildkite-pr-comment-->

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/16952)
* Commit: 59b18b57c7bb410f09ba8623b8c51b794206465a

### Failed CI Steps
* [Check integrations auditd_manager](https://buildkite.com/elastic/integrations/builds/16952#01926f6c-274d-45b6-9adf-e9d02dd04b5e)

### History
* :broken_heart: [Build #16929](https://buildkite.com/elastic/integrations/builds/16929) failed 87b75b09a8ad7969eb4a3033ee7c7c80b7e6a835
* :broken_heart: [Build #16821](https://buildkite.com/elastic/integrations/builds/16821) failed f1ab7f92ec70452727ce9b284dc6425af810ea41

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/16952","number":16952,"commit":"59b18b57c7bb410f09ba8623b8c51b794206465a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/16929","number":16929,"commit":"87b75b09a8ad7969eb4a3033ee7c7c80b7e6a835"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/16821","number":16821,"commit":"f1ab7f92ec70452727ce9b284dc6425af810ea41"}],"number":16952}
buildkite-pr-comment-->

mjwolf commented on packages/auditd_manager/data_stream/auditd/agent/stream/auditd.yml.hbs:
```diff
@@ -7,8 +7,17 @@ socket_type: '{{socket_type}}'
 failure_mode: {{failure_mode}}
-{{#if audit_rules}}
+{{#if session_data}}
+audit_rules: "{{escape_multiline_string audit_rules}}
```
I think you'll need to rework this to be like `processors` below so you can have both `session_data` and `audit_rules` and only one `audit_rules` block

I also got an `escape_multiline_string`, but I maybe didn't have the needed kibana changes

opauloh replied:
I could manage to have processors logic cleaner because it uses type yaml, so the UI validates the Yaml format, audit_rules uses textarea and its more flexible in the input, but trickier to concat.

> I also got an escape_multiline_string, but I maybe didn't have the needed kibana changes

Yes that was merged today on kibana main so it will probably be available in the next snapshot.

mjwolf replied:
Could you have `audit_rules:` outside the if statements?

It's ok to have an empty `audit_rules` block, but not have two `audit_rules` (auditbeat won't report any error, but the 2nd block just won't work, so it can cause a silent error), and it's important to be able to have both the session data rules and any other custom rules users add 

I haven't tried it, but maybe something like this will work:
```
audit_rules: |
    {{#if session_data}}
    "{{escape_multiline_string audit_rules}}
      {{escape_multiline_string "
      # Session data audit rules
      -a always,exit -F arch=b64 -S execve,execveat -k exec
      -a always,exit -F arch=b64 -S exit_group
      -a always,exit -F arch=b64 -S setsid"}}"
    {{/if}}
    {{#if audit_rules}}
    {{escape_string audit_rules}}
    {{/if}}
```

opauloh replied:
I see what you mean, I tried your solution but I got this `YAMLException`

```
YAMLException: can not read a block mapping entry; a multiline key may not be an implicit key (22:7)

 19 | -a always,exit -F arch=b64 -S open,tr ...
 20 |
 21 |
 22 |       # Session data audit rules
------------^
 23 |
 24 |       -a always,exit -F arch=b64 -S e ...
```

However, I adjusted the indentation to make sure once the handlebar file is parsed it doesn't include the key twice:

Those are the streams part of the `elastic-agent.yaml` for each scenario:


### Session data checked, Audit rules empty:

```yaml
streams:
      - id: >-
          audit/auditd-auditd_manager.auditd-c430e68e-fcde-4317-90f8-d5bccaf25e75
        type: audit/auditd
        data_stream:
          dataset: auditd_manager.auditd
          type: logs
        condition: ${host.platform} == 'linux'
        include_raw_message: true
        socket_type: ''
        immutable: false
        resolve_ids: true
        failure_mode: silent
        audit_rules: |-


          # Session data audit rules
          -a always,exit -F arch=b64 -S execve,execveat -k exec
          -a always,exit -F arch=b64 -S exit_group
          -a always,exit -F arch=b64 -S setsid
        backlog_limit: 8192
        rate_limit: 0
        include_warnings: false
        backpressure_strategy: auto
        tags:
          - auditd_manager-auditd
        processors:
          - add_session_metadata:
              backend: auto
```

### Audit rules filled, Session data unchecked

```yaml
streams:
      - id: >-
          audit/auditd-auditd_manager.auditd-5bc9f711-81fd-47cc-8bae-dbb44b9f981f
        type: audit/auditd
        data_stream:
          dataset: auditd_manager.auditd
          type: logs
        condition: ${host.platform} == 'linux'
        include_raw_message: true
        socket_type: ''
        immutable: false
        resolve_ids: true
        failure_mode: silent
        audit_rules: >-
          # Unauthorized access attempts to files (unsuccessful).

          -a always,exit -F arch=b32 -S
          open,creat,truncate,ftruncate,openat,open_by_handle_at -F exit=-EACCES
          -F auid>=1000 -F auid!=4294967295 -F key=access

          -a always,exit -F arch=b32 -S
          open,creat,truncate,ftruncate,openat,open_by_handle_at -F exit=-EPERM
          -F auid>=1000 -F auid!=4294967295 -F key=access

          -a always,exit -F arch=b64 -S
          open,truncate,ftruncate,creat,openat,open_by_handle_at -F exit=-EACCES
          -F auid>=1000 -F auid!=4294967295 -F key=access

          -a always,exit -F arch=b64 -S
          open,truncate,ftruncate,creat,openat,open_by_handle_at -F exit=-EPERM
          -F auid>=1000 -F auid!=4294967295 -F key=access
        backlog_limit: 8192
        rate_limit: 0
        include_warnings: false
        backpressure_strategy: auto
        tags:
          - auditd_manager-auditd
```

### Session data checked, audit rules filled

```yaml
streams:
      - id: >-
          audit/auditd-auditd_manager.auditd-62408e1a-806d-406f-aba8-0ae1506f15f6
        type: audit/auditd
        data_stream:
          dataset: auditd_manager.auditd
          type: logs
        condition: ${host.platform} == 'linux'
        include_raw_message: true
        socket_type: ''
        immutable: false
        resolve_ids: true
        failure_mode: silent
        audit_rules: >-
          # Unauthorized access attempts to files (unsuccessful).

          -a always,exit -F arch=b32 -S
          open,creat,truncate,ftruncate,openat,open_by_handle_at -F exit=-EACCES
          -F auid>=1000 -F auid!=4294967295 -F key=access

          -a always,exit -F arch=b32 -S
          open,creat,truncate,ftruncate,openat,open_by_handle_at -F exit=-EPERM
          -F auid>=1000 -F auid!=4294967295 -F key=access

          -a always,exit -F arch=b64 -S
          open,truncate,ftruncate,creat,openat,open_by_handle_at -F exit=-EACCES
          -F auid>=1000 -F auid!=4294967295 -F key=access

          -a always,exit -F arch=b64 -S
          open,truncate,ftruncate,creat,openat,open_by_handle_at -F exit=-EPERM
          -F auid>=1000 -F auid!=4294967295 -F key=access


          # Session data audit rules

          -a always,exit -F arch=b64 -S execve,execveat -k exec

          -a always,exit -F arch=b64 -S exit_group

          -a always,exit -F arch=b64 -S setsid
        backlog_limit: 8192
        rate_limit: 0
        include_warnings: false
        backpressure_strategy: auto
        tags:
          - auditd_manager-auditd
        processors:
          - add_session_metadata:
              backend: auto
```

### session unchecked, audit rules 

```yaml
streams:
      - id: >-
          audit/auditd-auditd_manager.auditd-d17efb9a-8968-49b7-b5a9-8a8377611659
        type: audit/auditd
        data_stream:
          dataset: auditd_manager.auditd
          type: logs
        condition: ${host.platform} == 'linux'
        include_raw_message: true
        socket_type: ''
        immutable: false
        resolve_ids: true
        failure_mode: silent
        backlog_limit: 8192
        rate_limit: 0
        include_warnings: false
        backpressure_strategy: auto
        tags:
          - auditd_manager-auditd

```

It should be working as expected now, no duplicated `audit_rules` in the yaml

mjwolf replied:
Thanks, that looks good to me, and the config in the agent looks good

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/16953)
* Commit: 3a026aac158ae1b74407ee2a7ab706a2161d665b

### History
* :broken_heart: [Build #16952](https://buildkite.com/elastic/integrations/builds/16952) failed 59b18b57c7bb410f09ba8623b8c51b794206465a
* :broken_heart: [Build #16929](https://buildkite.com/elastic/integrations/builds/16929) failed 87b75b09a8ad7969eb4a3033ee7c7c80b7e6a835
* :broken_heart: [Build #16821](https://buildkite.com/elastic/integrations/builds/16821) failed f1ab7f92ec70452727ce9b284dc6425af810ea41

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/16953","number":16953,"commit":"3a026aac158ae1b74407ee2a7ab706a2161d665b"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/16952","number":16952,"commit":"59b18b57c7bb410f09ba8623b8c51b794206465a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/16929","number":16929,"commit":"87b75b09a8ad7969eb4a3033ee7c7c80b7e6a835"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/16821","number":16821,"commit":"f1ab7f92ec70452727ce9b284dc6425af810ea41"}],"number":16953}
buildkite-pr-comment-->

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/17035)
* Commit: c7df0b46477945ac178752c52b2c7cd557ea6bdf

### History
* :green_heart: [Build #16953](https://buildkite.com/elastic/integrations/builds/16953) succeeded 3a026aac158ae1b74407ee2a7ab706a2161d665b
* :broken_heart: [Build #16952](https://buildkite.com/elastic/integrations/builds/16952) failed 59b18b57c7bb410f09ba8623b8c51b794206465a
* :broken_heart: [Build #16929](https://buildkite.com/elastic/integrations/builds/16929) failed 87b75b09a8ad7969eb4a3033ee7c7c80b7e6a835
* :broken_heart: [Build #16821](https://buildkite.com/elastic/integrations/builds/16821) failed f1ab7f92ec70452727ce9b284dc6425af810ea41

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/17035","number":17035,"commit":"c7df0b46477945ac178752c52b2c7cd557ea6bdf"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/16953","number":16953,"commit":"3a026aac158ae1b74407ee2a7ab706a2161d665b"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/16952","number":16952,"commit":"59b18b57c7bb410f09ba8623b8c51b794206465a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/16929","number":16929,"commit":"87b75b09a8ad7969eb4a3033ee7c7c80b7e6a835"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/16821","number":16821,"commit":"f1ab7f92ec70452727ce9b284dc6425af810ea41"}],"number":17035}
buildkite-pr-comment-->

opauloh commented:
I also added one integration in [this](https://auditd-manager.kb.us-west2.gcp.elastic-cloud.com) instance using the zip attached. Credentials [here](https://p.elstc.co/paste/kKS3+l15#-hMBaMfCzzTDVvXxBuPF0ZSVuUWPWDisqTmHl87bnfz) to access the instance.


### Screenshots

![image](https://github.com/user-attachments/assets/01785d5a-4d3e-4ccd-97f0-a3b330a3ccc4)

![image](https://github.com/user-attachments/assets/4564cf54-e107-4c3f-908d-8f9807ed0725)

![image](https://github.com/user-attachments/assets/ee91703e-0519-4cc2-87b0-c0fac3dce5bc)

### Agent logs

Agent logs shows that 8 of 8 audit rules was successfully added

```
14:40:40.012
elastic_agent
[elastic_agent][info] Elastic Agent started
14:40:40.266
elastic_agent
[elastic_agent][info] Starting upgrade watcher
14:40:40.267
elastic_agent
[elastic_agent][info] Upgrade Watcher invoked
14:40:40.270
elastic_agent
[elastic_agent][info] APM instrumentation disabled
14:40:40.271
elastic_agent
[elastic_agent][info] Gathered system information
14:40:40.326
elastic_agent
[elastic_agent][info] Detected available inputs and outputs
14:40:40.327
elastic_agent
[elastic_agent][info] Capabilities file not found in /opt/Elastic/Agent/capabilities.yml
14:40:40.327
elastic_agent
[elastic_agent][info] Determined allowed capabilities
14:40:40.328
elastic_agent
[elastic_agent][info] Loading baseline config from /opt/Elastic/Agent/elastic-agent.yml
14:40:40.424
elastic_agent
[elastic_agent][info] Upgrade Watcher started
14:40:40.424
elastic_agent
[elastic_agent][info] update marker not present at '/opt/Elastic/Agent/data'
14:40:40.577
elastic_agent
[elastic_agent][info] GRPC comms socket listening at localhost:6789
14:40:40.577
elastic_agent
[elastic_agent][info] Parsed configuration and determined agent is managed locally
14:40:40.579
elastic_agent
[elastic_agent][info] GRPC control socket listening at unix:///opt/Elastic/Agent/elastic-agent.sock
14:40:40.579
elastic_agent
[elastic_agent][info] Created control socket symlink /run/elastic-agent.sock -> /opt/Elastic/Agent/elastic-agent.sock; allowing unix:///run/elastic-agent.sock connection
14:40:40.581
elastic_agent
[elastic_agent][info] Starting grpc control protocol listener on port 6789 with max_message_size 104857600
14:40:40.581
elastic_agent
[elastic_agent][info] Configuration changes detected
14:40:40.583
elastic_agent
[elastic_agent][info] Source URI changed from "https://artifacts.elastic.co/downloads/" to "https://artifacts.elastic.co/downloads/"
14:40:40.583
elastic_agent
[elastic_agent][info] Docker provider skipped, unable to connect: Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
14:40:40.584
elastic_agent
[elastic_agent][info] Starting monitoring server with cfg &config.MonitoringConfig{Enabled:true, MonitorLogs:true, MonitorMetrics:true, MetricsPeriod:"", LogMetrics:true, HTTP:(*config.MonitoringHTTPConfig)(0xc0015f5920), Namespace:"default", Pprof:(*config.PprofConfig)(nil), MonitorTraces:false, APM:config.APMConfig{Environment:"", APIKey:"", SecretToken:"", Hosts:[]string(nil), GlobalLabels:map[string]string(nil), TLS:config.APMTLS{SkipVerify:false, ServerCertificate:"", ServerCA:""}}, Diagnostics:config.Diagnostics{Uploader:config.Uploader{MaxRetries:10, InitDur:1000000000, MaxDur:600000000000}, Limit:config.Limit{Interval:60000000000, Burst:1}}}
14:40:40.584
elastic_agent
[elastic_agent][info] creating monitoring API with cfg api.Config{Enabled:true, Host:"http://localhost:6791", Port:6791, User:"", SecurityDescriptor:"", Timeout:5000000000}
14:40:40.584
elastic_agent
[elastic_agent][info] Starting stats endpoint
14:40:40.584
elastic_agent
[elastic_agent][info] Metrics endpoint listening on: 127.0.0.1:6791 (configured: http://localhost:6791)
14:40:40.695
elastic_agent
[elastic_agent][info] Updating running component model
14:40:40.930
elastic_agent
[elastic_agent][info] component model updated
14:40:40.930
elastic_agent
[elastic_agent][info] Updating running component model
14:40:41.000
elastic_agent
system/metrics-default
[elastic_agent][info] Spawned new component system/metrics-default: Starting: spawned pid '105176'
14:40:41.000
elastic_agent
system/metrics-default
[elastic_agent][info] Spawned new unit system/metrics-default-unique-system-metrics-input: Starting: spawned pid '105176'
14:40:41.000
elastic_agent
system/metrics-default
[elastic_agent][info] Spawned new unit system/metrics-default: Starting: spawned pid '105176'
14:40:41.415
elastic_agent
[elastic_agent][info] control checkin v2 protocol has chunking enabled
14:40:41.416
elastic_agent
system/metrics-default
[elastic_agent][info] Component state changed system/metrics-default (STARTING->HEALTHY): Healthy: communicating with pid '105176'
14:40:41.915
elastic_agent
[elastic_agent][info] control checkin v2 protocol has chunking enabled
14:40:41.952
elastic_agent
[elastic_agent][info] reexec shutdown channel triggered
14:40:42.517
elastic_agent
[elastic_agent][info] Elastic Agent started
14:40:42.736
elastic_agent
[elastic_agent][info] Starting upgrade watcher
14:40:42.738
elastic_agent
[elastic_agent][info] Upgrade Watcher invoked
14:40:42.740
elastic_agent
[elastic_agent][info] APM instrumentation disabled
14:40:42.741
elastic_agent
[elastic_agent][info] Gathered system information
14:40:42.787
elastic_agent
[elastic_agent][info] Detected available inputs and outputs
14:40:42.787
elastic_agent
[elastic_agent][info] Capabilities file not found in /opt/Elastic/Agent/capabilities.yml
14:40:42.787
elastic_agent
[elastic_agent][info] Determined allowed capabilities
14:40:42.787
elastic_agent
[elastic_agent][info] Loading baseline config from /opt/Elastic/Agent/elastic-agent.yml
14:40:42.865
elastic_agent
[elastic_agent][info] Upgrade Watcher started
14:40:42.865
elastic_agent
[elastic_agent][info] update marker not present at '/opt/Elastic/Agent/data'
14:40:42.972
elastic_agent
[elastic_agent][info] GRPC comms socket listening at localhost:6789
14:40:42.991
elastic_agent
[elastic_agent][info] Parsed configuration and determined agent is managed by Fleet
14:40:42.992
elastic_agent
[elastic_agent][info] GRPC control socket listening at unix:///opt/Elastic/Agent/elastic-agent.sock
14:40:42.993
elastic_agent
[elastic_agent][info] Created control socket symlink /run/elastic-agent.sock -> /opt/Elastic/Agent/elastic-agent.sock; allowing unix:///run/elastic-agent.sock connection
14:40:42.993
elastic_agent
[elastic_agent][info] Starting grpc control protocol listener on port 6789 with max_message_size 104857600
14:40:42.995
elastic_agent
[elastic_agent][info] Docker provider skipped, unable to connect: Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
14:40:43.613
elastic_agent
[elastic_agent][info] Fleet gateway started
14:40:45.736
elastic_agent
[elastic_agent][info] Setting fallback log level <nil> from policy
14:40:45.755
elastic_agent
[elastic_agent][info] Source URI changed from "https://artifacts.elastic.co/downloads/" to "https://artifacts.elastic.co/downloads/"
14:40:45.755
elastic_agent
[elastic_agent][info] Starting monitoring server with cfg &config.MonitoringConfig{Enabled:true, MonitorLogs:true, MonitorMetrics:true, MetricsPeriod:"", LogMetrics:true, HTTP:(*config.MonitoringHTTPConfig)(0xc001bc0f00), Namespace:"default", Pprof:(*config.PprofConfig)(nil), MonitorTraces:true, APM:config.APMConfig{Environment:"", APIKey:"", SecretToken:"", Hosts:[]string(nil), GlobalLabels:map[string]string(nil), TLS:config.APMTLS{SkipVerify:false, ServerCertificate:"", ServerCA:""}}, Diagnostics:config.Diagnostics{Uploader:config.Uploader{MaxRetries:10, InitDur:1000000000, MaxDur:600000000000}, Limit:config.Limit{Interval:60000000000, Burst:1}}}
14:40:45.755
elastic_agent
[elastic_agent][info] creating monitoring API with cfg api.Config{Enabled:true, Host:"http://localhost:6791", Port:6791, User:"", SecurityDescriptor:"", Timeout:5000000000}
14:40:45.756
elastic_agent
[elastic_agent][info] Starting stats endpoint
14:40:45.756
elastic_agent
[elastic_agent][info] Metrics endpoint listening on: 127.0.0.1:6791 (configured: http://localhost:6791)
14:40:45.777
elastic_agent
[elastic_agent][info] Updating running component model
14:40:46.123
elastic_agent
system/metrics-default
[elastic_agent][info] Spawned new component system/metrics-default: Starting: spawned pid '105212'
14:40:46.123
elastic_agent
system/metrics-default
[elastic_agent][info] Spawned new unit system/metrics-default: Starting: spawned pid '105212'
14:40:46.123
elastic_agent
system/metrics-default
[elastic_agent][info] Spawned new unit system/metrics-default-system/metrics-system-3afa3385-f28c-42b3-a2f2-66097d044cf6: Starting: spawned pid '105212'
14:40:46.428
elastic_agent
[elastic_agent][info] control checkin v2 protocol has chunking enabled
14:40:46.516
elastic_agent
system/metrics-default
[elastic_agent][info] Component state changed system/metrics-default (STARTING->HEALTHY): Healthy: communicating with pid '105212'
14:40:46.760
elastic_agent
audit/auditd-default
[elastic_agent][info] Spawned new component audit/auditd-default: Starting: spawned pid '105221'
14:40:46.760
elastic_agent
audit/auditd-default
[elastic_agent][info] Spawned new unit audit/auditd-default-audit/auditd-auditd-0a42fa72-91ea-452a-83ce-4f8d19cd3592: Starting: spawned pid '105221'
14:40:46.760
elastic_agent
audit/auditd-default
[elastic_agent][info] Spawned new unit audit/auditd-default: Starting: spawned pid '105221'
14:40:46.983
elastic_agent.auditbeat
audit/auditd-default
[elastic_agent.auditbeat][info] Home path: [/opt/Elastic/Agent/data/elastic-agent-8.15.2-621bbc/components] Config path: [/opt/Elastic/Agent/data/elastic-agent-8.15.2-621bbc/components] Data path: [/opt/Elastic/Agent/data/elastic-agent-8.15.2-621bbc/run/audit/auditd-default] Logs path: [/opt/Elastic/Agent/data/elastic-agent-8.15.2-621bbc/components/logs]
14:40:46.988
elastic_agent.auditbeat
audit/auditd-default
[elastic_agent.auditbeat][info] Beat ID: 06bc7aca-77f2-45a5-8992-ab905313f5c2
14:40:46.993
elastic_agent.auditbeat
audit/auditd-default
[elastic_agent.auditbeat][info] Output reload is enabled, the beat will restart as needed on change of output config
14:40:46.993
elastic_agent.auditbeat
audit/auditd-default
[elastic_agent.auditbeat][info] Set gc percentage to: 100
14:40:46.995
elastic_agent.auditbeat
audit/auditd-default
[elastic_agent.auditbeat][info] running under elastic-agent, per-beat lockfiles disabled
14:40:46.995
elastic_agent.auditbeat
audit/auditd-default
[elastic_agent.auditbeat][info] Starting stats endpoint
14:40:46.996
elastic_agent.auditbeat
audit/auditd-default
[elastic_agent.auditbeat][info] Syscall filter successfully installed
14:40:46.996
elastic_agent.auditbeat
audit/auditd-default
[elastic_agent.auditbeat][info] Beat info
14:40:46.996
elastic_agent.auditbeat
audit/auditd-default
[elastic_agent.auditbeat][info] Build info
14:40:46.996
elastic_agent.auditbeat
audit/auditd-default
[elastic_agent.auditbeat][info] Go runtime info
14:40:46.998
elastic_agent.auditbeat
audit/auditd-default
[elastic_agent.auditbeat][info] Host info
14:40:46.999
elastic_agent.auditbeat
audit/auditd-default
[elastic_agent.auditbeat][info] Process info
14:40:46.999
elastic_agent.auditbeat
audit/auditd-default
[elastic_agent.auditbeat][info] Setup Beat: auditbeat; Version: 8.15.2
14:40:47.000
elastic_agent.auditbeat
audit/auditd-default
[elastic_agent.auditbeat][info] Metrics endpoint listening on: /opt/Elastic/Agent/data/tmp/3eljxlWKeJO3S9gwrrPPiNozl_k8bSCW.sock (configured: unix:///opt/Elastic/Agent/data/tmp/3eljxlWKeJO3S9gwrrPPiNozl_k8bSCW.sock)
14:40:47.001
elastic_agent.auditbeat
audit/auditd-default
[elastic_agent.auditbeat][error] add_cloud_metadata: received error failed requesting openstack metadata: Get "https://169.254.169.254/2009-04-04/meta-data/instance-id": dial tcp 169.254.169.254:443: connect: connection refused
14:40:47.002
elastic_agent.auditbeat
audit/auditd-default
[elastic_agent.auditbeat][error] add_cloud_metadata: received error failed with http status code 404
14:40:47.002
elastic_agent.auditbeat
audit/auditd-default
[elastic_agent.auditbeat][error] add_cloud_metadata: received error failed with http status code 404
14:40:47.002
elastic_agent.auditbeat
audit/auditd-default
[elastic_agent.auditbeat][error] add_cloud_metadata: received error failed with http status code 404
14:40:47.002
elastic_agent.auditbeat
audit/auditd-default
[elastic_agent.auditbeat][error] add_cloud_metadata: received error failed with http status code 404
14:40:47.003
elastic_agent.auditbeat
audit/auditd-default
[elastic_agent.auditbeat][error] SDK 2024/10/10 21:40:47 WARN falling back to IMDSv1: operation error ec2imds: getToken, http response error StatusCode: 405, request to EC2 IMDS failed
14:40:47.004
elastic_agent.auditbeat
audit/auditd-default
[elastic_agent.auditbeat][error] add_cloud_metadata: received error failed fetching EC2 Identity Document: operation error ec2imds: GetInstanceIdentityDocument, http response error StatusCode: 404, request to EC2 IMDS failed
14:40:47.005
elastic_agent.auditbeat
audit/auditd-default
[elastic_agent.auditbeat][info] Output is configured through Central Management
14:40:47.008
elastic_agent.auditbeat
audit/auditd-default
[elastic_agent.auditbeat][info] Beat name: csp-paulo-e2e-test-auditdmanager
14:40:47.008
elastic_agent.auditbeat
audit/auditd-default
[elastic_agent.auditbeat][info] Starting metrics logging every 30s
14:40:47.008
elastic_agent.auditbeat
audit/auditd-default
[elastic_agent.auditbeat][info] auditbeat start running.
14:40:47.023
elastic_agent
[elastic_agent][info] control checkin v2 protocol has chunking enabled
14:40:47.024
elastic_agent
audit/auditd-default
[elastic_agent][info] Component state changed audit/auditd-default (STARTING->HEALTHY): Healthy: communicating with pid '105221'
14:40:47.028
elastic_agent.auditbeat
audit/auditd-default
[elastic_agent.auditbeat][info] BeatV2Manager.unitListen UnitChanged.ID(audit/auditd-default), UnitChanged.Type(added), UnitChanged.Trigger(4): added/feature_change_triggered
14:40:47.028
elastic_agent.auditbeat
audit/auditd-default
[elastic_agent.auditbeat][info] BeatV2Manager.unitListen UnitChanged.ID(audit/auditd-default-audit/auditd-auditd-0a42fa72-91ea-452a-83ce-4f8d19cd3592), UnitChanged.Type(added), UnitChanged.Trigger(4): added/feature_change_triggered
14:40:47.033
elastic_agent.auditbeat
audit/auditd-default
[elastic_agent.auditbeat][info] add_cloud_metadata: hosting provider type detected as gcp, metadata={"cloud":{"account":{"id":"elastic-security-dev"},"availability_zone":"us-east1-c","instance":{"id":"2987215839827295004","name":"csp-paulo-e2e-test-auditdmanager"},"machine":{"type":"e2-medium"},"project":{"id":"elastic-security-dev"},"provider":"gcp","region":"us-east1","service":{"name":"GCE"}}}
14:40:47.118
elastic_agent
log-default
[elastic_agent][info] Spawned new component log-default: Starting: spawned pid '105229'
14:40:47.119
elastic_agent
log-default
[elastic_agent][info] Spawned new unit log-default-logfile-system-3afa3385-f28c-42b3-a2f2-66097d044cf6: Starting: spawned pid '105229'
14:40:47.119
elastic_agent
log-default
[elastic_agent][info] Spawned new unit log-default: Starting: spawned pid '105229'
14:40:47.392
elastic_agent
[elastic_agent][info] control checkin v2 protocol has chunking enabled
14:40:47.393
elastic_agent
log-default
[elastic_agent][info] Component state changed log-default (STARTING->HEALTHY): Healthy: communicating with pid '105229'
14:40:47.439
elastic_agent
system/metrics-default
[elastic_agent][info] Unit state changed system/metrics-default (STARTING->HEALTHY): Healthy
14:40:47.496
elastic_agent
system/metrics-default
[elastic_agent][info] Unit state changed system/metrics-default-system/metrics-system-3afa3385-f28c-42b3-a2f2-66097d044cf6 (STARTING->HEALTHY): Healthy
14:40:48.029
elastic_agent.auditbeat
audit/auditd-default
[elastic_agent.auditbeat][info] Applying performance preset 'balanced': {
  "bulk_max_size": 1600,
  "compression_level": 1,
  "idle_connection_timeout": "3s",
  "queue": {
    "mem": {
      "events": 3200,
      "flush": {
        "min_events": 1600,
        "timeout": "10s"
      }
    }
  },
  "worker": 1
}
14:40:48.030
elastic_agent.auditbeat
audit/auditd-default
[elastic_agent.auditbeat][warn] Performance preset 'balanced' overrides user setting for field 'bulk_max_size'
14:40:48.030
elastic_agent.auditbeat
audit/auditd-default
[elastic_agent.auditbeat][info] elasticsearch url: https://aa84255b4a5b437a86d024960ddcfe08.us-west2.gcp.elastic-cloud.com:443
14:40:48.031
elastic_agent
audit/auditd-default
[elastic_agent][info] Unit state changed audit/auditd-default (STARTING->HEALTHY): Healthy
14:40:48.033
elastic_agent.auditbeat
audit/auditd-default
[elastic_agent.auditbeat][info] auditd module is running as euid=0 on kernel=5.15.0-1067-gcp
14:40:48.038
elastic_agent.auditbeat
audit/auditd-default
[elastic_agent.auditbeat][info] socket_type=unicast will be used.
14:40:48.116
elastic_agent.auditbeat
audit/auditd-default
[elastic_agent.auditbeat][info] backfilled 111 processes
14:40:48.192
elastic_agent
[elastic_agent][info] control checkin v2 protocol has chunking enabled
14:40:48.403
elastic_agent
log-default
[elastic_agent][info] Unit state changed log-default (STARTING->HEALTHY): Healthy
14:40:48.435
elastic_agent
log-default
[elastic_agent][info] Unit state changed log-default-logfile-system-3afa3385-f28c-42b3-a2f2-66097d044cf6 (STARTING->HEALTHY): Healthy
14:40:48.679
elastic_agent
[elastic_agent][info] control checkin v2 protocol has chunking enabled
14:40:49.118
elastic_agent
[elastic_agent][info] control checkin v2 protocol has chunking enabled
14:40:56.879
elastic_agent.auditbeat
audit/auditd-default
[elastic_agent.auditbeat][info] backend=auto using ebpf
14:40:56.880
elastic_agent.auditbeat
audit/auditd-default
[elastic_agent.auditbeat][info] Deleted 4 pre-existing audit rules.
14:40:56.881
elastic_agent.auditbeat
audit/auditd-default
[elastic_agent.auditbeat][info] Successfully added 8 of 8 audit rules.
14:40:56.882
elastic_agent
audit/auditd-default
[elastic_agent][info] Unit state changed audit/auditd-default-audit/auditd-auditd-0a42fa72-91ea-452a-83ce-4f8d19cd3592 (STARTING->HEALTHY): Healthy
14:40:56.932
elastic_agent.auditbeat
audit/auditd-default
[elastic_agent.auditbeat][info] audit status from kernel at start
14:40:56.932
elastic_agent.auditbeat
audit/auditd-default
[elastic_agent.auditbeat][info] Setting kernel backlog wait time to prevent backpressure propagating to the kernel.
14:41:06.936
elastic_agent.auditbeat
audit/auditd-default
[elastic_agent.auditbeat][info] Connecting to backoff(elasticsearch(https://aa84255b4a5b437a86d024960ddcfe08.us-west2.gcp.elastic-cloud.com:443))
14:41:07.143
elastic_agent.auditbeat
audit/auditd-default
[elastic_agent.auditbeat][info] Attempting to connect to Elasticsearch version 8.16.0-SNAPSHOT (default)
14:41:07.819
elastic_agent.auditbeat
audit/auditd-default
[elastic_agent.auditbeat][info] Connection to backoff(elasticsearch(https://aa84255b4a5b437a86d024960ddcfe08.us-west2.gcp.elastic-cloud.com:443)) established
```


cc @mjwolf

mjwolf reviewed (approved):
LGTM!
